### PR TITLE
INT: Add `Use destructuring declaration` intention

### PR DESCRIPTION
**Pull request by mchernyavsky**
destructure1

### Update (destructure tuple):
destructure2

### CODE REVIEW

**Comment by vlad20012** [comment]
Hmm, why this works only for structs with named fields, but not also for tuple structs and regular structs?

**Comment by mchernyavsky** [comment]
@vlad20012 Fixed.

File: src/main/kotlin/org/rust/ide/intentions/DestructureIntention.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import org.rust.lang.core.types.ty.TyTuple
+import org.rust.lang.core.types.type
+
+class DestructureIntention : RsElementBaseIntentionAction<DestructureIntention.Context>() {
+    override fun getText(): String = "Use destructuring declaration"
+    override fun getFamilyName(): String = text
+
+    data class Context(
+        val patIdent: RsPatIdent,
+        val struct: RsStructItem? = null,
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/intentions/DestructureIntention.kt]: These fields can't be filled both, so consider use `sealed class`

File: src/main/kotlin/org/rust/ide/intentions/DestructureIntention.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    private fun handleTuple(project: Project, editor: Editor, patIdent: RsPatIdent, fieldNum: Int) {
+        val patTuple = RsPsiFactory(project).createPatTuple(fieldNum)
+        val newPatTuple = patIdent.replace(patTuple) as? RsPatTup ?: return
+        replacePatsWithPlaceholders(editor, newPatTuple)
+    }
+
+    companion object {
+        private fun replacePatsWithPlaceholders(editor: Editor, newPat: RsPat) {
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/intentions/DestructureIntention.kt]: I don't understand what this code does. so consider add a comment =)

**Comment by mchernyavsky** [comment]
@vlad20012 Fixed.

**Pull request merged**